mycity/mycity/utilities/arcgis_utils.py

```diff
@@ -0,0 +1,283 @@
+    body_as_string = formatted_tuple[0]
+    updated_header = formatted_tuple[1]
+    # POST request over network
+    response = _post_request(base_url, body_as_string, updated_header)
+
+    if response.status_code != 200:
+        logger.debug("Response Error: {}".format(str(response.status_code)))
+        return None
+    else:
+        response_json = response.json()
+        logger.debug("Response JSON: {}".format(str(response_json)))
+
+        routes = response_json['routes']
+        features = routes['features']
+        attributes = features[0]['attributes']
+        facility_id = attributes['FacilityID']
+        travel_time_in_minutes = attributes['Total_TravelTime']
+        travel_distance_in_miles = attributes['Total_Miles']
+
+        formatted_travel_time_in_minutes = _format_float(float(travel_time_in_minutes))
+        formatted_travel_distance_in_miles = _format_float(float(travel_distance_in_miles))
+
+        travel_time_string = "{} minutes".format(formatted_travel_time_in_minutes)
+        travel_distance_string = "{} miles".format(formatted_travel_distance_in_miles)
+
+        facility_key_index = int(facility_id) - 1
+        facility_key = facility_key_list[facility_key_index]
+        facility_address = destination_addresses[facility_key]
+
+        destination_dict = {
+                'Address': facility_address,
+                'Driving_time': travel_time_string,
+                'Driving_distance': travel_distance_string
+                }
+
+        logger.debug("Returning closest destination: {}".format(str(destination_dict)))
+        return destination_dict
+
+
+def format_multipart_form_request(url, params):
+    """
+    Formats a multipart/form POST request
+    for requests module properly for
+    ESRI ARCGis API
+
+    :param url: String containg URL of API
+    :param params: Dictionary of paramters
+    :return Two-Tuple containing 1) String representing
+        params to be passed as data to POST request
+        and 2) Dictionary with modified headers
+    """
+    logger.debug("URL: {}, Params: {}".format(url, str(params)))
+
+    updated_params = _modify_multipart_form_params(params)
+    req = requests.Request('POST', url, files=updated_params)
+    prepared_request = req.prepare()
+
+    updated_header = prepared_request.headers
+    updated_header.pop('Content-Length')
+    updated_header['cache-control'] = "no-cache"
+
+    # Change body from byte string to string
+    body_as_bytes = prepared_request.body
+    body_as_string = body_as_bytes.decode('utf-8')
+
+    return (body_as_string, updated_header)
+
+
+
+
+def _modify_multipart_form_params(params):
+    """
+    Helper function for _format_multipart_form_request
+    that properly formats param dictionary
+
+    :param params: Dictionary of request body paramters
+    :return Dictionary of updated request body parameters
+    """
+    updated_params = {}
+    for key in params:
+        value = params[key]
```

Comment:
Could do 
`
for key, value in params.items():
`

Reply:
Correct, this iterator is more concise